src/languages/asciidoc.js

```diff
@@ -104,7 +104,13 @@ export default function(hljs) {
         begin: '^(NOTE|TIP|IMPORTANT|WARNING|CAUTION):\\s+',
         relevance: 10
       },
-      // inline strong
+      // inline unconstrained strong
+      {
+        className: 'strong',
+        begin: /\*{2}/,
```

Comment:
Is it not possible to escape an * here?

Reply:
you can escape but I'm not sure I quite understand the existing rule:

```js
// allow escaped asterisk followed by word char
contains: [
  {
    begin: '\\\\*\\w',
    relevance: 0
  }
]
```

The following should not be highlighted:

```
this is a \*strong* word.
```

```
this is a str*\*on**g word.
```

But the last example is pretty uncommon (and fragile). In this case, it would be better to use substitution:

```
:stars: **
this is a str{stars}on{stars}g word.
```

Since the AsciiDoc specification is under-way, I would ignore edge cases for now because I'm not even sure what is the expected result.

Here's a few examples, with the input and the rendered result:

![escaped](https://user-images.githubusercontent.com/333276/99192944-0517c480-2776-11eb-8151-c4c0ee4ab759.png)

![more](https://user-images.githubusercontent.com/333276/99192943-047f2e00-2776-11eb-9dbe-cd0472b1f21e.png)

As you can see it's quite complex...

Reply:
> I'm not sure I quite understand the existing rule:

It's simply allowing \* inside of bold so that it doesn't trigger "end bold".  I'm not sure why the requirement of "one character after"... that might be specific to asciidoc.  I brought it up since we already handle this for unconstrained so it seems we should also for constrained unless for some reason it's only really used in one type but not the other.

> with the input and the rendered result:

We're less interested in the rendered result.  _We are not a rendering engine._  We're trying to highlight the semantics of the language.  So for some things that means strong won't be strong at all, it might be red or blue.

```
this is a str**o**n**g word.
```

This would be broken with your current PR as the `**g` would be bold (if we can find two linebreaks anywhere else).

```
this is a \*strong* word.
```

We'd need to add a general rule to eat a `\*` sequence then when it's found to avoid triggering any of the other rules.  This should be a test case no doubt.

---

We should really start here with which cases are reasonable to handle, add markup tests. Then see if we can make those tests green. :)

Reply:
The markup tests should include the negative cases too, to make sure we properly handle them, etc...

Reply:
> We should really start here with which cases are reasonable to handle, add markup tests. Then see if we can make those tests green. :)

💯 I will add tests.

> We're less interested in the rendered result. We are not a rendering engine. 

I know, it was just easier to share every variations.

> The markup tests should include the negative/edge cases too, to make sure we properly handle them, etc...

👍